src/main/java/org/opensearch/ad/cluster/ADDataMigrator.java

```diff
@@ -0,0 +1,288 @@
+                logger.info("AD job index doesn't exist, no need to migrate");
+                return;
+            }
+
+            if (detectionIndices.doesDetectorStateIndexExist()) {
+                migrateDetectorInternalStateToRealtimeTask();
+            } else {
+                // If detection index doesn't exist, create index and backfill realtime task.
+                detectionIndices.initDetectionStateIndex(ActionListener.wrap(r -> {
+                    if (r.isAcknowledged()) {
+                        logger.info("Created {} with mappings.", CommonName.DETECTION_STATE_INDEX);
+                        migrateDetectorInternalStateToRealtimeTask();
+                    } else {
+                        String error = "Create index " + CommonName.DETECTION_STATE_INDEX + " with mappings not acknowledged";
+                        logger.warn(error);
+                    }
+                }, e -> {
+                    if (ExceptionsHelper.unwrapCause(e) instanceof ResourceAlreadyExistsException) {
+                        migrateDetectorInternalStateToRealtimeTask();
+                    } else {
+                        logger.error("Failed to init anomaly detection state index", e);
+                    }
+                }));
+            }
+        }
+    }
+
+    /**
+     * Migrate detector internal state to realtime task.
+     */
+    public void migrateDetectorInternalStateToRealtimeTask() {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+            .query(new MatchAllQueryBuilder())
+            .size(MAX_DETECTOR_UPPER_LIMIT);
+        SearchRequest searchRequest = new SearchRequest(ANOMALY_DETECTOR_JOB_INDEX).source(searchSourceBuilder);
+        client.search(searchRequest, ActionListener.wrap(r -> {
+            if (r == null || r.getHits().getTotalHits() == null || r.getHits().getTotalHits().value == 0) {
+                logger.info("No anomaly detector job found, no need to migrate");
+                return;
+            }
+            ConcurrentLinkedQueue<AnomalyDetectorJob> detectorJobs = new ConcurrentLinkedQueue<>();
+            Iterator<SearchHit> iterator = r.getHits().iterator();
+            while (iterator.hasNext()) {
+                SearchHit searchHit = iterator.next();
+                try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, searchHit.getSourceRef())) {
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                    AnomalyDetectorJob job = AnomalyDetectorJob.parse(parser);
+                    detectorJobs.add(job);
+                } catch (IOException e) {
+                    logger.error("Fail to parse AD job " + searchHit.getId(), e);
+                }
+            }
+            logger.info("Total AD jobs to backfill realtime task: {}", detectorJobs.size());
+            backfillRealtimeTask(detectorJobs, true);
```

Comment:
Can you explain why there would be multiple jobs, and what the reason is to migrate multiple over, rather than just the latest?